service/history/workflow/mutable_state_impl.go

```diff
@@ -3188,21 +3232,88 @@ func (ms *MutableStateImpl) AddWorkflowExecutionUpdateAcceptedEvent(protocolInst
 		return nil, err
 	}
 	event := ms.hBuilder.AddWorkflowExecutionUpdateAcceptedEvent(protocolInstanceID, updAcceptance)
-	// TODO (alex-update): Async workflow update will require ReplicateWorkflowExecutionUpdateAcceptedEvent
-	// which restores update in the registry if it is not there.
+	if err := ms.ReplicateWorkflowExecutionUpdateAcceptedEvent(event); err != nil {
+		return nil, err
+	}
 	return event, nil
 }
 
+func (ms *MutableStateImpl) ReplicateWorkflowExecutionUpdateAcceptedEvent(
+	event *historypb.HistoryEvent,
+) error {
+	attrs := event.GetWorkflowExecutionUpdateAcceptedEventAttributes()
+	if attrs == nil {
+		return serviceerror.NewInternal("wrong event type in call to ReplicateworkflowExecutionUpdateAcceptedEvent")
+	}
+	ms.updateRecords[attrs.GetAcceptedRequest().GetMeta().GetUpdateId()] = &persistencespb.WorkflowExecutionUpdateRecord{
+		Value: &persistencespb.WorkflowExecutionUpdateRecord_AcceptancePointer{
+			AcceptancePointer: &historyspb.HistoryEventPointer{EventId: event.GetEventId()},
+		},
+	}
+	ms.writeEventToCache(event)
```

Comment:
There is comment inside `writeEventToCache` functions which describes why it is called for different type events. I think it is already outdated (I see more calls than cases described there) but still believe it is better to add comment about update events there.